hera_sim/visibilities/simulators.py

```diff
@@ -163,10 +159,18 @@ def validate(self):
             raise ValueError("The number of beams provided must be at least "
                              "as great as the greatest beam_id.")
 
-        if self.point_source_flux is not None:
-            if self.point_source_flux.shape[0] != self.sky_freqs.shape[0]:
-                raise ValueError("point_source_flux must have the same number "
-                                 "of freqs as sky_freqs.")
+        if (
+            self.point_source_flux is not None
+            and self.point_source_flux.shape[0] != self.sky_freqs.shape[0]
+        ):
+            if self.point_source_flux.shape[0] == 1:
+                self.point_source_flux = np.repeat(self.point_source_flux, self.sky_freqs.shape[0]).reshape((self.sky_freqs.shape[0], -1))
```

Comment:
Oh, nice! That sounds way better. So can I get a `SkyModel` instance with master-branch code now? You seemed to suggest it still returns recarray's until your PR is merged.

Reply:
Yeah, once my PR is merged it will still return recarrays, but will have the option to return a SkyModel instead. Eventually, I'd like it to return a SkyModel by default, but using the keyword makes it easier to transition.
Do we want this merged first, or my PR on pyuvsim? This seems to be required to fix Bryna's PR (which runs the hera_sim tests from pyuvsim). I'm pretty sure my PR won't break hera_sim as is. Maybe we can merge that first and then this branch can receive a SkyModel from `initialize_catalog_from_params` using the new keyword?

Reply:
Yeah, let's merge your PR first, then make this PR better long-term.

Reply:
I've played with this a bit, and updated it to use a `SkyModel` as the return type. I think we can do better in the future with a dedicated PR which updates the full machinery to use these `SkyModel`s, but it's at least using it a little now. However, running tests is giving me an error that when creating the `SkyModel` (using `obsparams`), it's using a float64 array for the frequencies, where it should be an astropy Quantity. Is that expected?

Reply:
Don't worry, it was me doing something stupid. Everything seems to work now.